test/core/codec/leb128/leb128_test.cpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "core/codec/leb128/leb128_algorithm_tester.hpp"
+#include "core/codec/leb128/leb128_codec_tester.hpp"
+
+using boost::math::tools::max_value;
+using boost::math::tools::min_value;
+using boost::multiprecision::uint128_t;
+using boost::multiprecision::uint256_t;
+using boost::multiprecision::uint512_t;
+using boost::multiprecision::uint1024_t;
+
+/**
+ * Support only unsigned values
+ */
+using supported_types = testing::Types<uint8_t,
+                                       uint16_t,
+                                       uint32_t,
+                                       uint64_t,
+                                       ::uint128_t,
+                                       ::uint256_t,
+                                       ::uint512_t,
+                                       ::uint1024_t>;
+
+/**
+ * LEB128 encode samples
+ * @info produced with https://github.com/Equim-chan/leb128
+ */
+std::vector<std::pair<uint128_t, std::string>> leb128_samples{
+    {8, "08"},
+    {120, "78"},
+    {8888, "b845"},
+    {667888, "f0e128"},
+    {624485, "e58e26"},
+    {4815162342, "e69786f811"},
+    {2432615184, "9086fb8709"},
+    {1234567890, "d285d8cc04"},
+    {9876543210, "eaadc0e524"},
+    {19283746574839301, "85dca084f2cea022"},
+    {9876543212345678903u, "b7b8a3a593f7a1888901"},
+    {uint128_t("123456789012345678901234567890"),
+     "d295fcf1e49df8b9c3edbfc8ee31"}};
```

Comment:
add tests for all supported types

Reply:
Added, samples was generated with github.com/fmoo/python-varint